src/main/java/com/radixdlt/engine/RadixEngine.java

```diff
@@ -42,27 +42,27 @@
 	private final ConstraintMachine constraintMachine;
 	private final CMStore virtualizedCMStore;
 	private final EngineStore<T> engineStore;
-	private final CopyOnWriteArrayList<AtomEventListener<T>> atomEventListeners = new CopyOnWriteArrayList<>();
-	private final CopyOnWriteArrayList<CMSuccessHook<T>> cmSuccessHooks = new CopyOnWriteArrayList<>();
+	private final AtomChecker<T> checker;
```

Comment:
Why the move away from hooks to a single checker?

Reply:
emphasis that checker is a real part of the checking pipeline and defines the engine rather than just a hook which can be added/removed.